domain/machine/service/provider.go

```diff
@@ -27,13 +30,15 @@ type Provider interface {
 	environs.BootstrapEnviron
 	environs.InstanceTypesFetcher
 	environs.InstancePrechecker
+	environs.LXDProfiler
 }
 
 // ProviderService provides the API for working with machines using the
 // underlying provider.
 type ProviderService struct {
 	Service
 
+	lxdProfileMu   sync.Mutex
```

Comment:
Although we state that there should only be one service running at a time, we don't actually guarantee it. There should be no state on a service, a mutex is state.

We should either push this down into the provider to handle this, or create a distributed lock against the container, similar to the following:

https://github.com/juju/juju/blob/49697cbd586c32b888048a2172805c35b8d0a9d7/internal/objectstore/base.go#L125-L140